### PR TITLE
[12.0] Move restrictions from stock_putaway_storage_type_strategy to base module

### PR DESCRIPTION
**Pull request by grindtildeath**
Original description:

The code added here was defined in `stock_putaway_storage_type_strategy` (#5).
However, we want these restrictions to be defined in the base module `stock_storage_type`.

Updates:

 - Dedicated tree view to edit package/location storage type relation
 - Rename fields to drop stock_ prefix
 - Small fixes (security group, demo data, view names, ...)

Depends on :
 - [ ] https://github.com/OCA/server-tools/pull/1738

### CODE REVIEW

File: stock_storage_type/models/stock_location_storage_type.py [hunk 1]
```diff
@@ -28,3 +29,68 @@ class StockLocationStorageType(models.Model):
...
+    # max_height = fields.Boolean(
+    #     help="If checked, moves to the destination location will only be "
+    #          "allowed if the packaging height is lower than this maximum."
+    # )
+    # max_height_value = fields.Float()
+    # max_volume = fields.Boolean(
+    #     help="If checked, moves to the destination location will only be "
+    #          "allowed if the packaging volume is lower than this maximum."
+    # )
+    # max_volume_value = fields.Float()
```
Comment (Tonow-c2c) [inline stock_storage_type/models/stock_location_storage_type.py]: they is a reason to keep this comment ?

File: stock_storage_type/models/stock_location_storage_type.py [hunk 2]
```diff
@@ -28,3 +29,68 @@ class StockLocationStorageType(models.Model):
         'stock_location_storage_type_id',
         'stock_package_storage_type_id',
     )
+
+    only_empty = fields.Boolean(
+        help="If checked, moves to the destination location will only be "
+             "allowed if there are not any existing quant nor planned move on "
+             "this location")
```
Comment (Tonow-c2c) [inline stock_storage_type/models/stock_location_storage_type.py:41]:
```suggestion
             "this location"
    )
```

File: stock_storage_type/models/stock_location_storage_type.py [hunk 3]
```diff
@@ -28,3 +29,68 @@ class StockLocationStorageType(models.Model):
...
+    def _check_empty_mix(self):
+        for location_storage_type in self:
+            if (
+                location_storage_type.only_empty and (
+                    location_storage_type.do_not_mix_lots or
+                    location_storage_type.do_not_mix_products
+                )
+            ):
+                raise ValidationError(_(
+                    "You cannot set 'Do not mix lots' or 'Do not mix products'"
```
Comment (Tonow-c2c) [inline stock_storage_type/models/stock_location_storage_type.py:72]: you can not rise 2 different error to separate `Do not mix lots` and `Do not mix products` and may be add the lot name in the errror

File: stock_storage_type/models/stock_location_storage_type.py [hunk 4]
```diff
@@ -28,3 +29,68 @@ class StockLocationStorageType(models.Model):
...
+
+    # @api.constrains('max_height', 'max_height_value', 'max_height_uom')
+    # def _check_max_height(self):
+    #     pass
+    #     # A value must be specified
+    #
+    # @api.constrains('max_volume', 'max_volume_value', 'max_volume_uom')
+    # def _check_max_volume(self):
+    #     pass
+    #     # A value must be specified
```
Comment (Tonow-c2c) [inline stock_storage_type/models/stock_location_storage_type.py]: still needed ?

**Review by jgrandguillaume — APPROVED** [review]
LGTM

**Review by damdam-s — APPROVED** [review]
lgtm
adding tests cases could help to maintain this

**Comment by jgrandguillaume** [comment]
As migrated to v 13, I close this one

**Pull request closed without merging**